src/net.c

```diff
@@ -174,6 +174,74 @@ static int addattr_l(struct nlmsghdr *n, int maxlen, int type, void *data, int a
 	return 0;
 }
 
+static int wait_for_nic(const char *nic_name)
+{
+	struct sockaddr_nl nls;
+	char buf[512];
+	int fd = -1;
+	int ret = -1;
+	char path[512];
+
+	sprintf(path, "/sys/class/net/%s/ifindex", nic_name);
+
+	if( access( path, F_OK ) != -1 ) {
+		fprintf(stdout, "test: nic %s exist\n", nic_name);
+	}
+
+	memset(&nls,0,sizeof(struct sockaddr_nl));
+	nls.nl_family = AF_NETLINK;
+	nls.nl_pid = getpid();
+	nls.nl_groups = -1;
+
+	fd = socket(PF_NETLINK, SOCK_DGRAM, NETLINK_KOBJECT_UEVENT);
+	if (fd==-1){
+		perror("Open socket failed\n");
+		return -1;
+	}
+
+	if( access( path, F_OK ) != -1 ) {
+		ret = 0;
+		fprintf(stdout, "nic %s exist\n", nic_name);
+		goto out;
+	}
+
+	if (bind(fd, (void *)&nls, sizeof(struct sockaddr_nl))) {
+		perror("Bind failed\n");
+		goto out;
+	}
+
+
+	while (1){
+		int  index = 0;
+		int  len   = 0;
+		char b[512];
+
+		len = recv(fd, buf, sizeof(buf), 0);
+		if (len == -1){
+			perror("fail to read from UEVENT netlink socket \n");
+			goto out;
+		}
+
+		while (index < len) {
+			index+= snprintf(b, sizeof(b),"%s", buf+index);
+			index++;
+			char *p = strstr(b, "DEVPATH=");
+			if (!p) { continue; }
+			printf("%s\n", p);
+			fprintf(stdout, "%s\n", p);
+			char *d = strstr(b, nic_name);
```

Comment:
@jcvenegas can you trigger off of the macaddress here? that is pretty unique. I think udev does provide macaddress info also